json/src/value.rs

```diff
@@ -1333,3 +1512,139 @@ impl<T: ?Sized> ToJson for T
         to_value(&self)
     }
 }
+
+/// Just like `std::borrow::Borrow<Value>` but borrowing may fail.
+pub trait TryBorrow<'a>: 'a + ::std::marker::Sized {
+    /// Returns reference to internal data if type matches.
+    fn try_borrow(value: &'a Value) -> Result<Self, TypeError>;
+}
+
+impl<'a> TryBorrow<'a> for &'a () {
+    fn try_borrow(value: &'a Value) -> Result<Self, TypeError> {
+        static EMPTY: () = ();
+        if value.is_null() { Ok(&EMPTY) } else { Err(TypeError { expected: ValueType::Null, provided: ValueType::of(value) }) }
+    }
+}
+
+impl<'a> TryBorrow<'a> for &'a bool {
```

Comment:
I was thinking about it too lately. I'm not sure how to define the macro to be readable enough. I was thinking about something like this:

`impl_try_borrows!(
   Bool => bool,
   I64 => i64,
// etc.
);`

What do you think?